### PR TITLE
#159987624 Add user signup feature

### PR DESCRIPTION
**Pull request by gonuoha**
#### What does this PR do?
Enable  user to signup

#### Description of Task to be completed?
- create user signup reducer
- create and render signup form
- post user details and display errors if any
- add tests for user signup

#### How should this be manually tested?
Run 'npm install'
Run 'npm run start'
view interface on the browser

#### What are the relevant pivotal tracker stories?
[#159987624](https://www.pivotaltracker.com/story/show/159987624)

### CODE REVIEW

File: src/actions/user.signup.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import axios from 'axios';
+
+import {
+    USER_SIGN_UP_FULFILLED,
+    USER_SIGN_UP_FAILED,
+    LOADING
+} from './action.types';
+
+const API = "https://thor-ah-staging.herokuapp.com";
+// const API = 'http://localhost:3001'
```
Comment (iamuchejude) [inline src/actions/user.signup.js]: Remove commented code on line 10

File: src/components/Header/AuthMenu.js [hunk 2]
```diff
@@ -4,29 +4,60 @@ import { Link } from 'react-router-dom';
...
+      modal = null;
+    }
+
     const { mobileCategory } = this.props;
     return (
       <div className={styles.links}>
         <ul>
           <li><Link to="/login">Login</Link></li>
-          <li><Link to="/register">Register</Link></li>
+          <button type='button' className={signupStyle.transparentBtn} onClick={this.toggleModal}><li>Register </li></button>
```
Comment (iamuchejude) [inline src/components/Header/AuthMenu.js]: You enclosed `<li>` tag in `<button>` element. I don't think it's right that way.
It's better you attach your event to `<Link>` or have your button enclosed in `<li>`

File: src/containers/Signup/SignUp.module.css [hunk 3]
```diff
@@ -0,0 +1,263 @@
+/* body,html{
```
Comment (iamuchejude) [inline src/containers/Signup/SignUp.module.css]: No commented code

File: src/tests/components/SocialSignup.test.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+import SocialSignup from '../../components/SocialSignup';
+
+describe('SocialSignup component', () => {
+    test("renders the SocialSignUp component", () => {
+        const wrapper = shallow(<SocialSignup />);
+
+        expect(wrapper.exists()).toBe(true);
+    });
+});
```
Comment (iamuchejude) [inline src/tests/components/SocialSignup.test.js]: Add a newline to this file

File: src/tests/containers/SignUp.test.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        const SignupContainer = shallow(<SignUp />);
+        expect(SignupContainer.exists()).toBe(true);
+    });
+
+    it('Throws error if required imputs to validated', () => {
+        const SignupContainer = shallow(<SignUp />);
+        expect(SignupContainer.exists()).toBe(true);
+    });
+
+    // it('should fail if no credentials are provided', () => {
```
Comment (iamuchejude) [inline src/tests/containers/SignUp.test.js]: Remove commented code

File: src/tests/reducers/user.reducer.test.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+    it('should handle USER_SIGN_UP_FAILED', () => {
+        const action = { type: types.USER_SIGN_UP_FAILED, payload: { email: 'email is already registered' } };
+        expect(signup({}, action)).toEqual({
+            user: {},
+            loading: false,
+            error: { email: 'email is already registered' },
+        });
+    });
+});
```
Comment (iamuchejude) [inline src/tests/reducers/user.reducer.test.js]: Add a new line to this file

**Review by iamuchejude — CHANGES_REQUESTED** [review]

**Review by iamuchejude — APPROVED** [review]

**Review by CEOehis — APPROVED** [review]
LGTM

**Review by d-beloved — APPROVED** [review]
LGTM

File: src/actions/signup.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import axios from 'axios';
+
+import {
+    SIGNUP_SUCCESS,
+    SIGNUP_ERROR,
+    LOADING
+} from './action.types';
```
Comment (johnkingzy) [inline src/actions/signup.js]: Let's have the all the action types store in a separate folder called `types`. Just the same way we the actions and reducers

**Pull request merged**